src/test/java/com/mathworks/ci/MatlabBuilderTest.java

```diff
@@ -279,4 +282,134 @@ public void verifyCustomCommandInvokedForBatchMode() throws Exception {
         jenkins.assertLogContains("-batch", build);
         jenkins.assertLogContains("runtests", build);
     }
+    
+    /*
+     * Test to verify if Automatic option passes appropriate test atrtifact values.
+     */
+
+    @Test
+    public void verifyRunTestAutomaticallyIsDefault() throws Exception {
+        this.matlabBuilder.setLocalMatlab(getMatlabroot("R2018b"));
+        RunTestsAutomaticallyOption runOption = new RunTestsAutomaticallyOption();
+        runOption.setTaCoberturaChkBx(true);
+        runOption.setTaJunitChkBx(true);
+        runOption.setTatapChkBx(true);
+        this.matlabBuilder.setTestRunTypeList(runOption);
+        project.getBuildersList().add(this.matlabBuilder);
+        FreeStyleBuild build = project.scheduleBuild2(0).get();
+        jenkins.assertLogContains("-batch", build);
+        jenkins.assertLogContains("true,true,true", build);
+    }
+    
+    /*
+     * Test to verify default value of getStringByName() when Automatic test mode. 
+     */
+
+    @Test
+    public void verifyDefaultValueOfgetStringByName() throws Exception {
+        this.matlabBuilder.setLocalMatlab(getMatlabroot("R2018b"));
+        RunTestsAutomaticallyOption runOption = new RunTestsAutomaticallyOption();    
+        Assert.assertNull(runOption.getStringByName("fakeChkBox"));
+    }
+    
+    /*
+     * Test to verify default value of getBooleanByName() when Custom test mode. 
+     */
+
+    @Test
+    public void verifyDefaultValueOfgetBooleanByName() throws Exception {
+        this.matlabBuilder.setLocalMatlab(getMatlabroot("R2018b"));
+        RunTestsWithCustomCommandOption runOption = new RunTestsWithCustomCommandOption();    
+        Assert.assertFalse(runOption.getBooleanByName("fakeCommand"));
+    }
+    
+    /*
+     * Test to verify when MATLAB version is older the R2017a
+     */
+
+    @Test
+    public void verifyMatlabVersionOlderThanR17a() throws Exception {
+        this.matlabBuilder.setLocalMatlab(getMatlabroot("R2018b"));
+        RunTestsAutomaticallyOption runOption = new RunTestsAutomaticallyOption();
+        runOption.setTaCoberturaChkBx(true);
+        runOption.setTaJunitChkBx(true);
+        runOption.setTatapChkBx(true);
+        this.matlabBuilder.setTestRunTypeList(runOption);
+        project.getBuildersList().add(this.matlabBuilder);
```

Comment:
Noticed that some of the code from this point to the end of the method is repeated in another test.  Could you pull this out into a different function?

Reply:
Done.